app/src/main/java/ru/gaket/themoviedb/presentation/movies/viewmodel/RxMoviesViewModelImpl.kt

```diff
@@ -0,0 +1,103 @@
+package ru.gaket.themoviedb.ru.gaket.themoviedb.presentation.movies.viewmodel
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import ru.gaket.themoviedb.model.movies.entities.Movie
+import ru.gaket.themoviedb.model.movies.repositories.MoviesRepository
+import ru.gaket.themoviedb.presentation.movies.viewmodel.MoviesResult
+import ru.gaket.themoviedb.ru.gaket.themoviedb.core.SchedulerProvider
+import ru.gaket.themoviedb.ru.gaket.themoviedb.presentation.movies.Navigator
+
+private const val DEBOUNCE_DELAY_TIME_MS = 500L
+
+internal class RxMoviesViewModelImpl(
+    schedulerProvider: SchedulerProvider,
+    private val moviesRepository: MoviesRepository,
+    private val navigator: Navigator,
+) : RxMoviesViewModel() {
+
+    // TODO #2: use one of the subjects to get new query from the text changes flow.
+    // Here it is enough to create publish subject
+    override val queryInput = TODO()
+
+    override val searchResultOutput = MutableLiveData<MoviesResult>()
+    override val searchStateOutput = MutableLiveData<SearchState>()
+
+    // TODO #3: create subscription class to collect all subscription you will do below.
+    // Reminder: use CompositeDisposable class
+
+    init {
+        queryInput
+        // TODO #4: in the init block we have to convert coroutines flow in MoviesViewModelImpl class to Rx.
```

Comment:
TODO 4 may sound as we need to update the whole flow at once. 
Let's leave the comment as is but move TODO #4 right before the place where we talk about debouncing